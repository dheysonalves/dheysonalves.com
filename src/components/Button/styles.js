import styled from 'styled-components';

export const Wrapper = styled.div`
`;

export const Button = styled.button`
	background-color: ${props => props.background || '#fff'};
	border: 2px solid #292F36;
	padding: 10px;
	width: 200px;
	cursor: pointer;
	outline: none;

	:disabled {
		opacity: 0.5;
	}
`;

export const LabelButton = styled.span`
	font-style: normal;
	font-weight: bold;
	font-size: 16px;
	line-height: 24px;
	text-transform: uppercase;
	color: #292F36;
`;
