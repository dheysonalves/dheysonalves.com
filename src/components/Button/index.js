import React from 'react';
import * as S from './styles';

const Button = ({ text = 'Submit' }) => {
	return (
		<S.Wrapper>
			<S.Button title="ok" disabled>
				<S.LabelButton>
					{text}
				</S.LabelButton>
			</S.Button>
		</S.Wrapper>
	)
}

export default Button;
