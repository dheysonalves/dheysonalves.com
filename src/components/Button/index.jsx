import React from 'react';
import * as S from './styles';

const Button = ({ text = 'Submit', click, ...props }) => {
  return (
    <S.Wrapper>
      <S.Button title="Wanna share an idea?" onClick={click} {...props}>
        <S.LabelButton>{text}</S.LabelButton>
      </S.Button>
    </S.Wrapper>
  );
};

export default Button;
