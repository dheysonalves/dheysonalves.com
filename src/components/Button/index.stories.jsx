import React from 'react';
import Button from '.';

export default {
    title: 'Components/Button',
    component: Button,
};

export const Default = (args) => <Button {...args} />;
