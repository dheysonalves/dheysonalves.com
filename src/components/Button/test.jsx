import React from 'react';
import { render, screen } from '@testing-library/react';

import Button from '.';

describe('<Button />', () => {
	it('should render the `Wanna share an idea?` text', () => {
		const { container } = render(<Button text="testing" />);

		expect(
			screen.getByRole('button', { name: /Wanna share an idea?/i })
		).toBeInTheDocument();
		expect(screen.getByRole('button')).not.toBeDisabled();
		expect(container.firstChild).toMatchSnapshot();
	});
});
