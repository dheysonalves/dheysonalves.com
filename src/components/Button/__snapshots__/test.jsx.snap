// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the \`Wanna share an idea?\` text 1`] = `
.c0 {
  background-color: #fff;
  border: 2px solid #292f36;
  padding: 10px;
  width: 200px;
  cursor: pointer;
  outline: none;
}

.c0:disabled {
  opacity: 0.5;
}

.c0:hover {
  background-color: #aeffd8;
}

.c0:active {
  background-color: #aeffc0;
}

.c0:focus {
  background-color: #aeffc0;
}

.c1 {
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 24px;
  text-transform: uppercase;
  color: #292f36;
}

<div
  class="styles__Wrapper-ox66x8-0"
>
  <button
    class="c0"
    role="button"
    title="Wanna share an idea?"
  >
    <span
      class="c1"
    >
      testing
    </span>
  </button>
</div>
`;
